ksp_plugin_test/flight_plan_test.cpp

```diff
@@ -0,0 +1,152 @@
+﻿#include "ksp_plugin/flight_plan.hpp"
+
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+namespace principia {
+
+using quantities::si::Kilogram;
+using quantities::si::Newton;
+using physics::BodyCentredNonRotatingDynamicFrame;
+
+namespace ksp_plugin {
+
+class FlightPlanTest : public testing::Test {
+ protected:
+  using TestNavigationFrame =
+      BodyCentredNonRotatingDynamicFrame<Barycentric, Navigation>;
+
+  FlightPlanTest() {
+    std::vector<not_null<std::unique_ptr<MassiveBody const>>> bodies;
+    bodies.emplace_back(
+        make_not_null_unique<MassiveBody>(1 * Pow<3>(Metre) / Pow<2>(Second)));
+    std::vector<DegreesOfFreedom<Barycentric>> initial_state{
+        {Barycentric::origin, Velocity<Barycentric>()}};
+    Instant const t0;
+    ephemeris_ = std::make_unique<Ephemeris<Barycentric>>(
+        std::move(bodies),
+        initial_state,
+        /*initial_time=*/t0 - 2 * π * Second,
+        integrators::McLachlanAtela1992Order5Optimal<Position<Barycentric>>(),
+        /*step=*/1 * Second,
+        /*fitting_tolerance=*/1 * Milli(Metre));
+    navigation_frame_ = std::make_unique<TestNavigationFrame>(
+        ephemeris_.get(),
+        ephemeris_->bodies().back());
+    root_.Append(t0_ - 2 * π * Second,
+                 {Barycentric::origin + Displacement<Barycentric>(
+                                            {1 * Metre, 0 * Metre, 0 * Metre}),
+                  Velocity<Barycentric>({0 * Metre / Second,
+                                         1 * Metre / Second,
+                                         0 * Metre / Second})});
+    root_.Append(t0_ + 2 * π * Second,
+                 {Barycentric::origin + Displacement<Barycentric>(
+                                            {1 * Metre, 0 * Metre, 0 * Metre}),
+                  Velocity<Barycentric>({0 * Metre / Second,
+                                         1 * Metre / Second,
+                                         0 * Metre / Second})});
+    flight_plan_ = std::make_unique<FlightPlan>(
+        &root_,
+        /*initial_time=*/t0_,
+        /*final_time=*/t0_ + 1.5 * Second,
+        /*initial_mass=*/1 * Kilogram,
+        ephemeris_.get(),
+        integrators::DormandElMikkawyPrince1986RKN434FM<
+            Position<Barycentric>>(),
+        /*length_integration_tolerance=*/ 1 * Milli(Metre),
+        /*speed_integration_tolerance=*/ 1 * Milli(Metre) / Second);
+  }
+
+  Instant const t0_;
+  std::unique_ptr<TestNavigationFrame> navigation_frame_;
+  std::unique_ptr<Ephemeris<Barycentric>> ephemeris_;
+  DiscreteTrajectory<Barycentric> root_;
+  std::unique_ptr<FlightPlan> flight_plan_;
+};
+
+TEST_F(FlightPlanTest, Append) {
+  auto const first_burn = [this]() -> Burn {
```

Comment:
Why are these lambdas and not plain variables?  I don't see that anything changes between the various invocations.